tests/slsa_analyzer/package_registry/test_npm_registry.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Tests for the npm registry."""
+
+import os
+from pathlib import Path
+
+import pytest
+
+from macaron.config.defaults import load_defaults
+from macaron.errors import ConfigurationError
+from macaron.slsa_analyzer.build_tool.base_build_tool import BaseBuildTool
+from macaron.slsa_analyzer.package_registry.npm_registry import NPMAttestationAsset, NPMRegistry
+
+
+def _create_npm_registry() -> NPMRegistry:
+    """Create an npm registry instance.
+
+    This private function is used as parameter for Hypothesis, which otherwise does not accept a fixture.
+    """
+    return NPMRegistry(
+        hostname="registry.npmjs.org", attestation_endpoint="-/npm/v1/attestations", request_timeout=20, enabled=True
+    )
+
+
+@pytest.fixture(name="npm_registry")
+def create_npm_registry() -> NPMRegistry:
+    """Create an npm registry instance."""
+    return _create_npm_registry()
+
+
+def test_disable_npm_registry(npm_registry: NPMRegistry, tmp_path: Path) -> None:
+    """Test disabling npm registry."""
+    config = """
+    [package_registry.npm]
+    enabled = False
+    """
+    config_path = os.path.join(tmp_path, "test_config.ini")
+    with open(config_path, mode="w", encoding="utf-8") as config_file:
+        config_file.write(config)
+    load_defaults(config_path)
+    npm_registry.load_defaults()
+
+    assert npm_registry.enabled is False
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        """
+            [package_registry.npm]
+            hostname =
+            """,
+        """
+            [package_registry.npm]
+            attestation_endpoint =
+            """,
+        """
+            [package_registry.npm]
+            request_timeout = foo
+            """,
+    ],
+)
+def test_npm_registry_invalid_config(npm_registry: NPMRegistry, tmp_path: Path, config: str) -> None:
+    """Test loading invalid npm registry configuration."""
+    config_path = os.path.join(tmp_path, "test_config.ini")
+    with open(config_path, mode="w", encoding="utf-8") as config_file:
+        config_file.write(config)
+    load_defaults(config_path)
+    with pytest.raises(ConfigurationError):
+        npm_registry.load_defaults()
+
+
+@pytest.mark.parametrize(
+    (
+        "build_tool_name",
+        "expected",
+    ),
+    [
+        ("npm", True),
+        ("yarn", True),
+        ("go", False),
+        ("maven", False),
+    ],
+)
+def test_is_detected(
+    npm_registry: NPMRegistry, build_tools: dict[str, BaseBuildTool], build_tool_name: str, expected: bool
+) -> None:
+    """Test that the registry is correctly detected for a build tool."""
+    npm_registry.load_defaults()
+    assert npm_registry.is_detected(build_tool=build_tools[build_tool_name]) == expected
+
+
+@pytest.mark.parametrize(
+    (
+        "namespace",
+        "artifact_id",
+        "version",
+        "expected",
```

Comment:
Do you think `expected` is the best name here? It is not the expected value of any `assert`, but rather the path path in the URL. I'm thinking `package_path` or something similar may be a better name.

Reply:
I agree. Or if we want to keep `expected`, the value could change to the exact HTTPS URL that we finally compare too.

Reply:
I prefer to the name `expected` as it is what the test expects to see as part of the `assert` statement.